### ISSUE TITLE
Fix false positives for `ui-*` generic system font keywords in font-family-name-quotes

### ISSUE THREAD

**Issue by nlydv** [stylelint/stylelint#6016]
### What steps are needed to reproduce the bug?

1. Set the `font-family-name-quotes` rule to `always-unless-keyword`
2. Add one the following generic keywords in a `font-family` property:
    - `ui-serif`
    - `ui-sans-serif`
    - `ui-monospace`
    - `ui-rounded`
3. Run Stylelint on the file

### What Stylelint configuration is needed to reproduce the bug?

```json
{
    "rules": {
        "font-family-name-quotes": "always-unless-keyword"
    }
}
```

### How did you run Stylelint?

CLI with `stylelint style.css --config .stylelintrc`

### Which version of Stylelint are you using?

`14.7.0`

### What did you expect to happen?

Stylelint would recognize `ui-monospace` et al. as a generic keyword that doesn't require quoting.

### What actually happened?

A warning was given stating that it expected quotes around `ui-monospace`.

### Does the bug relate to non-standard syntax?

No (see: https://www.w3.org/TR/css-fonts-4/#ui-serif-def and https://caniuse.com/extended-system-fonts).

### Proposal to fix the bug

Add `ui-serif`, `ui-sans-serif`, `ui-monospace`, and `ui-rounded` to the list of generic font-family keywords such that they be recognized by rules such as `font-family-name-quotes` and any other relevant rules.

**Comment by jeddy3** [comment]
@nlydv Thanks for the report, using the template and for linking to the specs.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

It'll just be case of updating our reference list:

https://github.com/stylelint/stylelint/blob/91c36013d5b904471f2f32adb307431e13d283c6/lib/reference/keywordSets.js#L93-L100
